pkg/controller/deployment/deployment_controller.go

```diff
@@ -599,6 +605,24 @@ func (dc *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControl
 	return true, err
 }
 
+func (dc *DeploymentController) cleanupOldRcs(controllers []*api.ReplicationController, deployment extensions.Deployment) error {
+	diff := len(controllers) - *deployment.Spec.RevisionHistoryLimit
+	if diff <= 0 {
```

Comment:
RevisionHistoryLimit should not include the current/latest RC.

If RevisionHistoryLimit is 1, and there are both a current and previous RC, we shouldn't delete any.